cvat/apps/engine/tests/test_lazy_list.py

```diff
@@ -0,0 +1,166 @@
+
+    def test_reverse(self):
+        self.lazy_list.reverse()
+        self.assertEqual(self.lazy_list, [3, 2, 1])
+
+    def test_sort(self):
+        unsorted_list = LazyList(string="3,1,2", converter=int)
+        unsorted_list.sort()
+        self.assertEqual(unsorted_list, [1, 2, 3])
+
+    def test_clear(self):
+        self.lazy_list.clear()
+        self.assertEqual(len(self.lazy_list), 0)
+
+    def test_index(self):
+        self.assertEqual(self.lazy_list.index(2), 1)
+
+    def test_count(self):
+        self.assertEqual(self.lazy_list.count(2), 1)
+
+    def test_setitem(self):
+        self.lazy_list[0] = 4
+        self.assertEqual(self.lazy_list[0], 4)
+
+    def test_delitem(self):
+        del self.lazy_list[0]
+        self.assertEqual(self.lazy_list, [2, 3])
+
+    def test_contains(self):
+        self.assertIn(2, self.lazy_list)
+
+    def test_reversed(self):
+        self.assertEqual(list(reversed(self.lazy_list)), [3, 2, 1])
+
+    def test_mul(self):
+        self.assertEqual(self.lazy_list * 2, [1, 2, 3, 1, 2, 3])
+
+    def test_rmul(self):
+        self.assertEqual(2 * self.lazy_list, [1, 2, 3, 1, 2, 3])
+
+    def test_imul(self):
+        self.lazy_list *= 2
+        self.assertEqual(self.lazy_list, [1, 2, 3, 1, 2, 3])
+
+    def test_extend(self):
+        self.lazy_list.extend([4, 5])
+        self.assertEqual(self.lazy_list, [1, 2, 3, 4, 5])
+
+    def test_add(self):
+        new_list = self.lazy_list + [4, 5]
+        self.assertEqual(new_list, [1, 2, 3, 4, 5])
+
+    def test_eq(self):
+        self.assertTrue(self.lazy_list == [1, 2, 3])
+
+    def test_iadd(self):
+        self.lazy_list += [4, 5]
+        self.assertEqual(self.lazy_list, [1, 2, 3, 4, 5])
+
+    def test_gt(self):
+        self.assertTrue(self.lazy_list > [1, 2])
+
+    def test_ge(self):
+        self.assertTrue(self.lazy_list >= [1, 2, 3])
+
+    def test_lt(self):
+        self.assertTrue(self.lazy_list < [1, 2, 3, 4])
+
+    def test_le(self):
+        self.assertTrue(self.lazy_list <= [1, 2, 3])
+
+    def test_lazy_list_with_lazy_list(self):
+        other_lazy_list = LazyList(string="4,5,6", converter=int)
+        combined_list = self.lazy_list + other_lazy_list
+        self.assertEqual(combined_list, [1, 2, 3, 4, 5, 6])
+
+    def test_pickle_support(self):
+        pickled = pickle.dumps(self.lazy_list)
+        unpickled = pickle.loads(pickled)
+        self.assertEqual(unpickled, [1, 2, 3])
+        self.assertEqual(unpickled.string, "")
+        self.assertTrue(unpickled.parsed)
+
+    def test_parse_before_accessing_decorator(self):
+        lazy_list_copy = LazyList(string="1,2,3", converter=int)
+        lazy_list_copy.append(4)
+        self.assertEqual(lazy_list_copy, [1, 2, 3, 4])
+
+    def test_parse_both_before_accessing_decorator(self):
+        other_list = LazyList(string="4,5", converter=int)
+        result = self.lazy_list + other_list
+        self.assertEqual(result, [1, 2, 3, 4, 5])
+
```

Comment:
It makes sense to add a test for calling `len()` in the middle of the first iteration.
```
for x in lazy_list:
    len(lazy_list)
```